docs/conf.py

```diff
@@ -26,6 +26,7 @@
 import os
 import sys
 
+sys.path.insert(0, os.path.abspath(".."))
```

Comment:
This could have been a separate PR right as overall it makes other types references as clickable links whether the type of a parameter, a base class a return type etc. I guess its ok here, though as a improvement to the docs its something that can go to stable to improve the current docs.

Reply:
I've extracted this out into #783 in order to allow a simple backporting of it, too :+1: